Tests/TestingApps/PaywallsTester/PaywallsTester/Configuration.swift

```diff
@@ -6,34 +6,64 @@
 //
 
 import Foundation
+import RevenueCat
 
-enum Configuration {
+class Configuration: ObservableObject {
+    static let shared = Configuration()
 
-    #warning("Configure API key if you want to test paywalls from your dashboard")
+    // This is modified by CI:
+    private static let apiKeyFromCIForTesting = ""
+    private static let apiKeyFromCIForDemos = ""
 
-    // Note: you can leave this empty to use the production server, or point to your own instance.
-    static let proxyURL = ""
-    static let apiKey = ""
+    @Published private(set) var currentMode: Mode
 
-    static let entitlement = "pro"
+    let entitlement = "pro"
 
-}
+    #warning("Configure API key if you want to test paywalls from your dashboard")
+    // Note: you can leave this empty to use the production server, or point to your own instance.
+    private let proxyURL = ""
+    private let apiKey = ""
 
-extension Configuration {
+    enum Mode {
+        case custom, testing, demos
+    }
 
-    static var effectiveApiKey: String = {
-        return Self.apiKey.nonEmpty ?? Self.apiKeyFromCI
-    }()
+    private init() {
+        currentMode = apiKey.isEmpty ? .testing : .custom
+    }
 
-    // This is modified by CI:
-    private static let apiKeyFromCI = ""
 
-}
+    var currentAPIKey: String {
+        switch currentMode {
+        case .custom:
+            self.apiKey
+        case .testing:
+            Self.apiKeyFromCIForTesting
+        case .demos:
+            Self.apiKeyFromCIForDemos
+        }
+    }
 
-// MARK: - Extensions
+    func reconfigure(for mode: Mode) {
+        self.currentMode = mode
+        Purchases.configure(
+            with: .init(withAPIKey: currentAPIKey)
+                .with(entitlementVerificationMode: .informational)
+                .with(usesStoreKit2IfAvailable: true)
+        )
+    }
 
-private extension String {
+    func configure() {
+        Purchases.logLevel = .verbose
+        Purchases.proxyURL = self.proxyURL.isEmpty
+        ? nil
+        : URL(string: self.proxyURL)!
 
-    var nonEmpty: String? { return self.isEmpty ? nil : self }
+        Purchases.configure(
+            with: .init(withAPIKey: currentAPIKey)
+                .with(entitlementVerificationMode: .informational)
+                .with(usesStoreKit2IfAvailable: true)
+        )
```

Comment:
This is duplicated in `reconfigure`

Reply:
yeah... this being 2 lines of simple code I didn't really think it'd be worth it to create another method and more indirection to avoid duplication, but I don't mind it. Adding.